### DISCUSSION TITLE
Global pooling with pre-defined nodes

### DISCUSSION THREAD

**Discussion by kimnupjuk** [pyg-team/pytorch_geometric#4904] [Q&A] [answered]
Hello, I want to do global pooling with pre-defined specific nodes for each graph.

However, implemented global pooling layer doesn't support pooling with index, is there any simple way to do that?

I found there wasn't any similar discussion on how to solve this.

Thank you!

**Comment by rusty1s** [comment, marked as answer]
You would first need to `mask/index_select` all the relevant nodes, and then perform global pooling on this subset:
```python
mask = ... nodes to pool
global_mean_pool(x[mask], batch[mask])
```
